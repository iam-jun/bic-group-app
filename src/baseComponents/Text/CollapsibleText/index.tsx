import React, {
  FC, memo, useCallback, useMemo, useState,
} from 'react';
import {
  View,
  StyleProp,
  ViewStyle,
  TouchableWithoutFeedback,
  StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import { isEqual } from 'lodash';
import { useBaseHook } from '~/hooks';
import Text, { TextProps } from '~/baseComponents/Text';

import MarkdownView from '~/beinComponents/MarkdownView';
import Markdown from '~/beinComponents/Markdown';
import CopyableView from '../../../beinComponents/CopyableView';
import { escapeMarkDown } from '~/utils/formatter';
import spacing from '~/theme/spacing';
import { TrackingEventContentReadProperties } from '~/services/tracking/Interface';
import { PostType } from '~/interfaces/IPost';
import { trackEvent } from '~/services/tracking';
import { TrackingEventContentReadAction, TrackingEventType } from '~/services/tracking/constants';

export interface CollapsibleTextProps extends TextProps {
  testID?: string;
  style?: StyleProp<ViewStyle>;
  content: string;
  limitLength?: number;
  shortLength?: number;
  toggleOnPress?: boolean;
  useMarkdown?: boolean;
  useMarkdownIt?: boolean;
  limitMarkdownTypes?: boolean;
  parentCommentId?: string;
  copyEnabled?: boolean;
  mentions?: any;
  BottomRightComponent?: React.ReactNode | React.ReactElement;
  isTracking?: boolean;

  onPress?: () => void;
  onPressAudience?: (audience: any, e?: any) => any;
  onToggleShowTextContent?: () => void;
  [x: string]: any;
}

const _CollapsibleText: FC<CollapsibleTextProps> = ({
  testID,
  style,
  content,
  mentions,
  limitLength = 120,
  shortLength = 120,
  toggleOnPress,
  useMarkdown,
  useMarkdownIt,
  limitMarkdownTypes,
  parentCommentId,
  copyEnabled,
  BottomRightComponent,
  isTracking,
  onPress,
  onPressAudience,
  onToggleShowTextContent,
  ...textProps
}: CollapsibleTextProps) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const { t } = useBaseHook();

  const [contentShowAll, setContentShowAll] = useState(false);

  const shortContent = useMemo(() => getShortContent(content, limitLength, shortLength), [content]);
  let _content = shortContent;
  if (!shortContent || contentShowAll) {
    _content = content;
  }
  const escapeMarkDownContent = useMemo(() => escapeMarkDown(content), [content]);

  const textTestID = parentCommentId
    ? 'collapsible_text.level_2.content'
    : 'collapsible_text.level_1.content';

  const _onToggleShowTextContent = useCallback(() => {
    const toggleContentShowAll = !contentShowAll;
    setContentShowAll(toggleContentShowAll);
    onToggleShowTextContent?.();

    // tracking event
    if (toggleContentShowAll && isTracking) {
      const eventContentReadProperties: TrackingEventContentReadProperties = {
        content_type: PostType.POST,
        action: TrackingEventContentReadAction.SEE_MORE,
      };
      trackEvent({ event: TrackingEventType.CONTENT_READ, properties: eventContentReadProperties });
    }
  }, [contentShowAll]);

  const _onPress = () => {
    if (onPress) {
      onPress();
    } else if (toggleOnPress) {
      _onToggleShowTextContent();
    }
  };

  const renderShortContent = () => {
    if (!shortContent) return null;

    return (
      <View style={styles.row}>
        <Text.SubtitleM
          testID="collapsible_text.short_content"
          onPress={_onToggleShowTextContent}
          color={colors.neutral60}
        >
          {contentShowAll
            ? t('common:text_see_less')
            : t('common:text_see_more')}
        </Text.SubtitleM>
        {BottomRightComponent}
      </View>
    );
  };

  const renderContentWithMarkdown = useCallback(() => (
    <View style={style}>
      {useMarkdownIt ? (
        <MarkdownView
          {...textProps}
          limitMarkdownTypes={limitMarkdownTypes}
          onPressAudience={onPressAudience}
        >
          {_content}
        </MarkdownView>
      ) : (
        <Markdown
          {...textProps}
          value={_content}
          mentions={mentions}
          textTestID={textTestID}
          limitMarkdownTypes={limitMarkdownTypes}
          onPressAudience={onPressAudience}
        />
      )}
      {renderShortContent()}
    </View>
  ), [mentions, _content]);

  const renderContent = () => (
    <View style={style}>
      <Text testID="collapsible_text.content" {...textProps}>
        {_content}
      </Text>
      {renderShortContent()}
    </View>
  );

  const WrapperComponent = copyEnabled
    ? CopyableView
    : TouchableWithoutFeedback;
  const disabled = !(onPress || (toggleOnPress && shortContent));

  return (
    <WrapperComponent
      testID={testID}
      activeOpacity={0.6}
      content={escapeMarkDownContent}
      disabled={disabled}
      onPress={_onPress}
    >
      {useMarkdown ? renderContentWithMarkdown() : renderContent()}
    </WrapperComponent>
  );
};

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'space-between',
    paddingVertical: spacing.padding.base,
  },
});

function propsAreEqual(
  prev: any, next: any,
) {
  return isEqual(
    prev, next,
  );
}

const CollapsibleText = memo(_CollapsibleText, propsAreEqual);
CollapsibleText.whyDidYouRender = true;
export default CollapsibleText;

const getShortContent = (c: string, limitLength: number, shortLength: number) => {
  if (!c) return '';

  if (c && c.length > limitLength) {
    return `${c.substring(
      0, shortLength,
    )}...`;
  }
  return '';
};
