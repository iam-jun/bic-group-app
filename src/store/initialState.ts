import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
