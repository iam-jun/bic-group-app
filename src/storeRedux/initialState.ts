import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
