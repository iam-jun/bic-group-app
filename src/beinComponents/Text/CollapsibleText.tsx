import React, {FC, memo, useEffect, useState} from 'react';
import {
  View,
  StyleProp,
  ViewStyle,
  TouchableWithoutFeedback,
} from 'react-native';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import {useBaseHook} from '~/hooks';
import Text, {TextProps} from '~/beinComponents/Text';

import MarkdownView from '~/beinComponents/MarkdownView';
import Markdown from '~/beinComponents/Markdown';
import CopyableView from '../CopyableView';
import {escapeMarkDown} from '~/utils/formatData';

export interface CollapsibleTextProps extends TextProps {
  testID?: string;
  style?: StyleProp<ViewStyle>;
  content: string;
  limitLength?: number;
  shortLength?: number;
  toggleOnPress?: boolean;
  useMarkdown?: boolean;
  useMarkdownIt?: boolean;
  limitMarkdownTypes?: boolean;
  parentCommentId?: string;
  copyEnabled?: boolean;
  onPress?: () => void;
  onPressAudience?: (audience: any, e?: any) => any;
  [x: string]: any;
}

const _CollapsibleText: FC<CollapsibleTextProps> = ({
  testID,
  style,
  content,
  limitLength = 120,
  shortLength = 120,
  toggleOnPress,
  useMarkdown,
  useMarkdownIt,
  limitMarkdownTypes,
  parentCommentId,
  copyEnabled,
  onPress,
  onPressAudience,
  ...textProps
}: CollapsibleTextProps) => {
  const getShortContent = (c?: string) => {
    if (c && c?.length > limitLength) {
      return `${c.substr(0, shortLength)}...`;
    } else {
      return '';
    }
  };

  const [contentShowAll, setContentShowAll] = useState(false);
  const [shortContent, setShortContent] = useState(getShortContent(content));

  const theme: ExtendedTheme = useTheme() as ExtendedTheme;
  const {colors} = theme;

  const {t} = useBaseHook();

  useEffect(() => {
    const newShort = getShortContent(content);
    if (newShort !== shortContent) {
      setShortContent(newShort);
    }
  }, [content]);

  const onToggleShowLess = () => setContentShowAll(!contentShowAll);

  const _onPress = () => {
    if (onPress) {
      onPress();
    } else if (toggleOnPress) {
      onToggleShowLess();
    }
  };

  const renderContentWithMarkdown = () => {
    return (
      <View style={style}>
        {useMarkdownIt ? (
          <MarkdownView
            {...textProps}
            limitMarkdownTypes={limitMarkdownTypes}
            onPressAudience={onPressAudience}>
            {!shortContent ? content : contentShowAll ? content : shortContent}
          </MarkdownView>
        ) : (
          <Markdown
            {...textProps}
            textTestID={
              parentCommentId
                ? 'collapsible_text.level_2.content'
                : 'collapsible_text.level_1.content'
            }
            onPressAudience={onPressAudience}
            value={
              !shortContent ? content : contentShowAll ? content : shortContent
            }
          />
        )}

        {!!shortContent && (
          <Text.BodyM
            testID="collapsible_text.markdown.short_content"
            onPress={onToggleShowLess}
            color={colors.gray50}>
            {contentShowAll
              ? t('common:text_see_less')
              : t('common:text_see_more')}
          </Text.BodyM>
        )}
      </View>
    );
  };

  const renderContent = () => {
    return (
      <Text style={style}>
        <Text testID="collapsible_text.content" {...textProps}>
          {!shortContent ? content : contentShowAll ? content : shortContent}
        </Text>
        {!!shortContent && (
          <Text.BodyM
            testID="collapsible_text.show_text"
            onPress={onToggleShowLess}
            color={colors.gray50}>
            {contentShowAll
              ? t('common:text_see_less')
              : t('common:text_see_more')}
          </Text.BodyM>
        )}
      </Text>
    );
  };

  const WrapperComponent = copyEnabled
    ? CopyableView
    : TouchableWithoutFeedback;

  return (
    <WrapperComponent
      testID={testID}
      activeOpacity={0.6}
      content={escapeMarkDown(content)}
      disabled={!(onPress || (toggleOnPress && shortContent))}
      onPress={_onPress}>
      {useMarkdown ? renderContentWithMarkdown() : renderContent()}
    </WrapperComponent>
  );
};

const CollapsibleText = memo(_CollapsibleText);
CollapsibleText.whyDidYouRender = true;
export default CollapsibleText;
