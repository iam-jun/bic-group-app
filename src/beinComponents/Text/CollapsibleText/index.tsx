import React, {
  FC, memo, useCallback, useMemo, useState,
} from 'react';
import {
  View,
  StyleProp,
  ViewStyle,
  TouchableWithoutFeedback,
  StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import { useBaseHook } from '~/hooks';
import Text, { TextProps } from '~/beinComponents/Text';

import MarkdownView from '~/beinComponents/MarkdownView';
import Markdown from '~/beinComponents/Markdown';
import CopyableView from '../../CopyableView';
import { escapeMarkDown } from '~/utils/formatData';
import spacing from '~/theme/spacing';

export interface CollapsibleTextProps extends TextProps {
  testID?: string;
  style?: StyleProp<ViewStyle>;
  content: string;
  limitLength?: number;
  shortLength?: number;
  toggleOnPress?: boolean;
  useMarkdown?: boolean;
  useMarkdownIt?: boolean;
  limitMarkdownTypes?: boolean;
  parentCommentId?: string;
  copyEnabled?: boolean;
  BottomRightComponent?: React.ReactNode | React.ReactElement;

  onPress?: () => void;
  onPressAudience?: (audience: any, e?: any) => any;
  onToggleShowTextContent?: () => void;
  [x: string]: any;
}

const _CollapsibleText: FC<CollapsibleTextProps> = ({
  testID,
  style,
  content,
  limitLength = 120,
  shortLength = 120,
  toggleOnPress,
  useMarkdown,
  useMarkdownIt,
  limitMarkdownTypes,
  parentCommentId,
  copyEnabled,
  BottomRightComponent,
  onPress,
  onPressAudience,
  onToggleShowTextContent,
  ...textProps
}: CollapsibleTextProps) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const { t } = useBaseHook();

  const [contentShowAll, setContentShowAll] = useState(false);

  const shortContent = useMemo(() => getShortContent(content, limitLength, shortLength), [content]);

  const _content = !shortContent ? content : contentShowAll ? content : shortContent;

  const _onToggleShowTextContent = useCallback(() => {
    setContentShowAll(!contentShowAll);
    onToggleShowTextContent?.();
  }, [contentShowAll]);

  const _onPress = () => {
    if (onPress) {
      onPress();
    } else if (toggleOnPress) {
      _onToggleShowTextContent();
    }
  };

  const renderShortContent = () => {
    if (!shortContent) return null;

    return (
      <View style={styles.row}>
        <Text.SubtitleM
          testID="collapsible_text.short_content"
          onPress={_onToggleShowTextContent}
          color={colors.neutral60}
        >
          {contentShowAll
            ? t('common:text_see_less')
            : t('common:text_see_more')}
        </Text.SubtitleM>
        {BottomRightComponent}
      </View>
    );
  };

  const renderContentWithMarkdown = () => (
    <View style={style}>
      {useMarkdownIt ? (
        <MarkdownView
          {...textProps}
          limitMarkdownTypes={limitMarkdownTypes}
          onPressAudience={onPressAudience}
        >
          {_content}
        </MarkdownView>
      ) : (
        <Markdown
          {...textProps}
          textTestID={
            parentCommentId
              ? 'collapsible_text.level_2.content'
              : 'collapsible_text.level_1.content'
          }
          limitMarkdownTypes={limitMarkdownTypes}
          onPressAudience={onPressAudience}
          value={_content}
        />
      )}
      {renderShortContent()}
    </View>
  );

  const renderContent = () => (
    <View style={style}>
      <Text testID="collapsible_text.content" {...textProps}>
        {_content}
      </Text>
      {renderShortContent()}
    </View>
  );

  const WrapperComponent = copyEnabled
    ? CopyableView
    : TouchableWithoutFeedback;

  return (
    <WrapperComponent
      testID={testID}
      activeOpacity={0.6}
      content={escapeMarkDown(content)}
      disabled={!(onPress || (toggleOnPress && shortContent))}
      onPress={_onPress}
    >
      {useMarkdown ? renderContentWithMarkdown() : renderContent()}
    </WrapperComponent>
  );
};

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'space-between',
    paddingVertical: spacing.padding.base,
  },
});

const CollapsibleText = memo(_CollapsibleText);
CollapsibleText.whyDidYouRender = true;
export default CollapsibleText;

const getShortContent = (c: string, limitLength: number, shortLength: number) => {
  if (!c) return '';

  if (c && c.length > limitLength) {
    return `${c.substring(
      0, shortLength,
    )}...`;
  }
  return '';
};
