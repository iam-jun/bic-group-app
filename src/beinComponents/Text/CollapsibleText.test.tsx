import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import initialState from '~/store/initialState';

import CollapsibleText from '~/beinComponents/Text/CollapsibleText';
import {
  fireEvent,
  renderWithRedux,
  configureStore,
  languages,
} from '~/test/testUtils';

afterEach(cleanup);

describe('Collapsible Text component', () => {
  const mockStore = configureStore([]);

  const description
    = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s. When an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting. Remaining essentially unchanged. It was popularised in the 1960s simply dummy text printing standard dummy text.\nLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.';

  const mentionName = '@vantest1';
  const commentData = {
    data: {
      mentions: {
        users: {
          vantest1: {
            data: {
              avatar:
                'https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png',
              fullname: 'vantest1',
              username: 'vantest1',
            },
          },
        },
      },
    },
  };

  it('renders correctly', () => {
    const rendered = render(<CollapsibleText content={description} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders correctly testID', () => {
    const rendered = render(
      <CollapsibleText testID="collapsible_text" content={description} />,
    );
    const { getByTestId } = rendered;
    const collapsibleTextComponent = getByTestId('collapsible_text');
    expect(collapsibleTextComponent).toBeDefined();
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render short text with a short length and limit the length', () => {
    const rendered = render(
      <CollapsibleText
        shortLength={50}
        limitLength={50}
        content={description}
      />,
    );
    const { getByTestId } = rendered;
    const shortTextComponent = getByTestId('collapsible_text.content');
    expect(shortTextComponent.props.children.length).toBe(50 + 3);
    const shortTextButtonComponent = getByTestId('collapsible_text.show_text');
    expect(shortTextButtonComponent.props.children).toBe(
      languages.common.text_see_more,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render full text when click read more button text ', () => {
    const rendered = render(<CollapsibleText content={description} />);
    const { getByTestId } = rendered;
    const btnSeeMore = getByTestId('collapsible_text.show_text');
    fireEvent.press(btnSeeMore);
    expect(btnSeeMore.props.children).toBe(languages.common.text_see_less);
    const fullTextComponent = getByTestId('collapsible_text.content');
    expect(fullTextComponent.props.children.length).toBe(description.length);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with markdown', () => {
    const storeData = { ...initialState };

    // @ts-ignore
    storeData.post.allComments.abc = commentData;
    const store = mockStore(storeData);

    const rendered = renderWithRedux(
      <CollapsibleText
        useMarkdown
        content={mentionName}
        selector="post.allComments.abc.data.mentions.users"
      />,
      store,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should call props onPressAudience', () => {
    const onPressAudience = jest.fn();

    const storeData = { ...initialState };

    // @ts-ignore
    storeData.post.allComments.abc = commentData;
    const store = mockStore(storeData);

    const rendered = renderWithRedux(
      <CollapsibleText
        useMarkdown
        content={mentionName}
        onPressAudience={onPressAudience}
        selector="post.allComments.abc.data.mentions.users"
      />,
      store,
    );

    const btnAudience = rendered.getByTestId('text_mention');
    expect(btnAudience).toBeDefined();
    fireEvent.press(btnAudience);
    expect(onPressAudience).toBeCalled();
  });

  it('should call props onPress', () => {
    const onPress = jest.fn();

    const rendered = render(
      <CollapsibleText
        testID="collapsible_text"
        content={description}
        onPress={onPress}
      />,
    );

    const collapsibleTextBtn = rendered.getByTestId('collapsible_text');
    expect(collapsibleTextBtn).toBeDefined();
    fireEvent.press(collapsibleTextBtn);
    expect(onPress).toBeCalled();
  });
});
