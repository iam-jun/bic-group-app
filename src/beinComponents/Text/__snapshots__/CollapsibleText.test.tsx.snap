// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible Text component renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2F3945",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.content"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard d...
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#007AFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.show_text"
    >
      Read more
    </Text>
  </Text>
</View>
`;

exports[`Collapsible Text component renders correctly testID 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="collapsible_text"
>
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2F3945",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.content"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard d...
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#007AFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.show_text"
    >
      Read more
    </Text>
  </Text>
</View>
`;

exports[`Collapsible Text component renders correctly with markdown 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <Text>
        <Text
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#386fe5",
                "fontFamily": "OpenSans-SemiBold",
              },
            ]
          }
          testID="text_mention"
        >
          @vantest1
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Collapsible Text component should render full text when click read more button text  1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2F3945",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.content"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s. When an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting. Remaining essentially unchanged. It was popularised in the 1960s simply dummy text printing standard dummy text.
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#007AFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.show_text"
    >
      Show less
    </Text>
  </Text>
</View>
`;

exports[`Collapsible Text component should render short text with a short length and limit the length 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2F3945",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.content"
    >
      Lorem Ipsum is simply dummy text of the printing a...
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#007AFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
      testID="collapsible_text.show_text"
    >
      Read more
    </Text>
  </Text>
</View>
`;
